src/pages/SigninPage/index.tsx

```diff
@@ -1,46 +1,98 @@
+import { PrintLineCommand } from '@/types/shell';
 
 import { footerStyle, headerStyle, layoutStyle } from './style';
 
 const SigninPage: FC = () => {
   const { push } = useHistory();
 
-  const requestSignin = async (body: LoginRequestBody) => {
+  const [
+    signinScript,
+    addPrintLineCommand,
+    addReadLineCommand,
+    createPrintLineCommand,
+  ] = useShellScript();
+
+  const checkUserName = async (val: string) => {
+    if (val.length) return { status: 'success', nextSequence: 2 } as const;
+
+    const errorLine = createPrintLineCommand(1, 'error', 'Please enter your ID');
+    return { status: 'error', ...errorLine } as const;
+  };
+
+  const checkPassword = async (val: string) => {
+    if (val.length) return { status: 'success', nextSequence: 3 } as const;
+
+    const errorLine = createPrintLineCommand(2, 'error', 'Please enter your Password');
+    return { status: 'error', ...errorLine } as const;
+  };
+
+  const checkSigninForm = async (
+    val: string,
+    body: LoginRequestBody,
+  ): Promise<
+    | {
+        status: 'success';
+        nextSequence: number | null;
+      }
+    | ({
+        status: 'error';
+      } & PrintLineCommand)
+  > => {
+    if (val !== 'y') {
+      const errorLine = createPrintLineCommand(1, 'error', `Access denied for user ${body.userId}`);
+      return { status: 'error', ...errorLine } as const;
+    }
+
     try {
       await authApi.login(body);
       push(Uri.service);
     } catch (error) {
-      if (!error.response) throw new Error(NETWORK_ERROR);
+      if (!error.response) {
+        const errorLine = createPrintLineCommand(1, 'error', NETWORK_ERROR);
+        return { status: 'error', ...errorLine } as const;
+      }
 
       const { status } = error.response;
-      if (loginError[status]) throw new Error(loginError[status]);
+      if (loginError[status]) {
+        const errorLine = createPrintLineCommand(1, 'error', loginError[status]);
+        return { status: 'error', ...errorLine } as const;
+      }
 
-      throw new Error(UNEXPECTED_ERROR);
+      const errorLine = createPrintLineCommand(1, 'error', UNEXPECTED_ERROR);
+      return { status: 'error', ...errorLine } as const;
     }
+
+    return { status: 'success', nextSequence: null };
   };
 
+  addPrintLineCommand(0, 1, '', 'telbby init v0.1.0');
+  addReadLineCommand(1, 'question','username', checkUserName, null, 'userId', 30);
+  addReadLineCommand(2, 'question', 'password', checkPassword, null, 'password', 35);
+  addReadLineCommand(3, 'question', 'Sign in? [y/n]', checkSigninForm, 'y', null, 1);
+
   return (
     <div css={layoutStyle}>
       <header css={headerStyle}>
         <Logo onlyImg width="70px" />
         <Jumbotron title="Sign in" />
       </header>
       <main>
-        <Shell
-          type="signin"
-          requestWhenQueryDone={requestSignin}
-          width="789px"
-          height="247px"
-        />
+        <Shell script={signinScript} width="789px" height="247px" />
```

Comment:
```suggestion
        <Shell width="789px" height="247px" >
          <ShellLine script={firstScript} write={true} />
        </Shell>
```

이런식으로 Shell 내부에서 하는 것보단 외부에서 ShellLine을 children으로 넣어주는 것은 어떨까요 ?
그리고 Shell 입장에서는 children의 타입을 shellLine으로 막아두면 좋을 것 같습니다.

Reply:
이렇게 할 경우 `Shell ` 내부 로직들이 바깥으로 드러나게 돼 쉘을 사용하는 모든 페이지에서 같은 로직을 재생성해야 할 것으로 예상됩니다 ㅠㅠ 
다만 주신 의견이 색달라 다양한 방면으로 생각하게 되는 것 같네요!! 참고하여 다시 개발해도록 하겠습니다 감사합니다 ㅎㅎㅎ!!